crates/miden-proving-service-client/src/lib.rs

```diff
@@ -13,12 +13,7 @@ extern crate std;
 use thiserror::Error;
 
 #[cfg(feature = "tx-prover")]
-pub mod generated;
-
-#[cfg(feature = "tx-prover")]
-mod prover;
-#[cfg(feature = "tx-prover")]
-pub use prover::RemoteTransactionProver;
+pub mod prover;
 
 /// Protobuf definition for the Miden proving service
 pub const SERVICE_PROTO: &str = include_str!("../proto/api.proto");
```

Comment:
Not from this PR, but `api.proto` sounds pretty generic. Should we rename this to something like `tx_prover.proto`?

Reply:
Yes, makes sense. specially now that we know that we're going to support multiple prover types. 
I opened #1108